### PR TITLE
PIDCMD

### PR DESCRIPTION
**Pull request by lycaon7**

### CODE REVIEW

File: src/main/java/frc/robot/commands/GenericPIDCMD.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.PIDSS;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+public class GenericPIDCMD extends CommandBase {
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/GenericPIDCMD.java:9]: change to PIDCMD? debate with @laviRZ

File: src/main/java/frc/robot/commands/GenericPIDCMD.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public GenericPIDCMD(PIDSS subsystem, DoubleSupplier setpoint) {
+        this(subsystem, setpoint, () -> false);
+    }
+
+    public GenericPIDCMD(PIDSS subsystem, DoubleSupplier setpoint, BooleanSupplier isFinished) {
+        this.subsystem = subsystem;
+        this.setpoint = setpoint;
+        this.isFinished = isFinished;
+        addRequirements(subsystem);
+    }
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/GenericPIDCMD.java:23]: complicated above simple constructor

File: src/main/java/frc/robot/subsystems/PIDSS.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package frc.robot.subsystems;
+
+public interface PIDSS extends MovableSubsystem {
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/PIDSS.java:3]: change to PIDSubsystem

File: src/main/java/frc/robot/subsystems/loader/LoaderSS.java [hunk 4]
```diff
@@ -19,8 +18,12 @@ public void move(double power) {
...
 
-    public void setVelocity(double velocity) {
-        motor.setSetpoint(velocity);
+    /**
+     * @param velocityRPM to be set to the motors in RPM
+     */
+    @Override
+    public void setSetpoint(double velocityRPM) {
+        motor.setSetpoint(velocityRPM);
     }
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/loader/LoaderSS.java:27]: change velocityRPM to `velocity`

File: src/main/java/frc/robot/subsystems/shooter/ShooterSS.java [hunk 5]
```diff
@@ -46,5 +46,6 @@ public double getVelocityRPM() {
     public double[] getValues() {
         return new double[] {masterMotor.getSelectedSensorPosition()};
     }
+
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:49]: delete blank line in line 49

**Review by Mboesn — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/loader/LoaderSS.java [hunk 6]
```diff
@@ -19,7 +18,11 @@ public void move(double power) {
         motor.set(power);
     }
 
-    public void setVelocity(double velocity) {
+    /**
+     * @param velocity to be set to the motors in RPM
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/loader/LoaderSS.java:22]: @ param velocity desired velocity in RPM

File: src/main/java/frc/robot/subsystems/loader/LoaderSS.java [hunk 7]
```diff
@@ -19,7 +18,11 @@ public void move(double power) {
...
+     */
+    @Override
+    public void setSetpoint(double velocity) {
         motor.setSetpoint(velocity);
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/loader/LoaderSS.java:26]: Conversions.RPMToFalcon(velocity, 1)

File: src/main/java/frc/robot/subsystems/loader/LoaderSS.java [hunk 8]
```diff
@@ -19,7 +18,11 @@ public void move(double power) {
...
     }
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/loader/LoaderSS.java:28]: edit get velocity to Conversions.FalconToRPM(velocity, 1)
and add docs

File: src/main/java/frc/robot/subsystems/shooter/ShooterSS.java [hunk 9]
```diff
@@ -19,10 +18,11 @@ public ShooterSS() {
     }
 
     /**
-     * @param velocityRPM to be set to the motors
+     * @param velocity to be set to the motors in RPM
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:21]: @ param velocity desired velocity in RPM

**Review by Mboesn — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/PIDCommand.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+public class PIDCommand extends CommandBase {
+    private final PIDSubsystem subsystem;
+    private final BooleanSupplier isFinished;
+    private final DoubleSupplier setpoint;
+
+    public PIDCommand(PIDSubsystem subsystem, DoubleSupplier setpoint, BooleanSupplier isFinished) {
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/PIDCommand.java:14]: docs

File: src/main/java/frc/robot/subsystems/PIDSubsystem.java [hunk 11]
```diff
@@ -0,0 +1,6 @@
+package frc.robot.subsystems;
+
+public interface PIDSubsystem extends MovableSubsystem {
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/PIDSubsystem.java:3]: docs

File: src/main/java/frc/robot/subsystems/shooter/ShooterSS.java [hunk 12]
```diff
@@ -47,4 +47,3 @@ public double getVelocityRPM() {
         return new double[] {masterMotor.getSelectedSensorPosition()};
     }
 }
-
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:50]: why last line deleted

**Review by Mboesn — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/intake/IntakeOpenerSS.java [hunk 13]
```diff
@@ -36,7 +37,8 @@ public double getAngle() {
      *
      * @param degree desired angle in degrees
      */
-    public void moveToAngle(double degree) {
+    @Override
+    public void setSetpoint(double degree) {
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/intake/IntakeOpenerSS.java:41]: rename parameter to setpoint

File: src/main/java/frc/robot/subsystems/loader/LoaderSS.java [hunk 14]
```diff
@@ -19,12 +19,19 @@ public void move(double power) {
         motor.set(power);
     }
 
-    public void setVelocity(double velocity) {
-        motor.setSetpoint(velocity);
+    /**
+     * @param velocity desired velocity in RPM
+     */
+    @Override
+    public void setSetpoint(double velocity) {
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/loader/LoaderSS.java:26]: rename parameter to setpoint

File: src/main/java/frc/robot/subsystems/loader/LoaderSS.java [hunk 15]
```diff
@@ -19,12 +19,19 @@ public void move(double power) {
...
+        motor.setSetpoint(Conversions.RPMToFalcon(velocity));
     }
 
+    /**
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/loader/LoaderSS.java:30]: Returns...

File: src/main/java/frc/robot/subsystems/shooter/ShooterSS.java [hunk 16]
```diff
@@ -19,10 +18,11 @@ public ShooterSS() {
     }
 
     /**
-     * @param velocityRPM to be set to the motors
+     * @param velocity desired velocity in RPM
      */
-    public void setVelocityRPM(double velocityRPM) {
-        masterMotor.setSetpoint(Conversions.RPMToFalcon(velocityRPM));
+    @Override
+    public void setSetpoint(double velocity) {
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:24]: rename parameter to setpoint

**Review by laviRZ — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/PIDCommand.java [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+        this.subsystem = subsystem;
+        this.setpoint = setpoint;
+        this.isFinished = isFinished;
+        addRequirements(subsystem);
+    }
+
+    public PIDCommand(PIDSubsystem subsystem, DoubleSupplier setpoint) {
```
[the remark this answers is not in the document]
Comment (lycaon7) [reply src/main/java/frc/robot/commands/PIDCommand.java:21]: git add codereview /.gitignore

File: src/main/java/frc/robot/subsystems/shooter/ShooterSS.java [hunk 18]
```diff
@@ -18,11 +18,11 @@ public ShooterSS() {
     }
 
     /**
-     * @param velocity desired velocity in RPM
+     * @param setpoint desired velocity in RPM
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:21]: the desired

File: src/main/java/frc/robot/subsystems/pitcher/PitcherSS.java [hunk 19]
```diff
@@ -36,14 +36,14 @@ public double getAngle() {
     /**
      * Sets the intake to the given angle
      *
-     * @param degree desired angle in degrees
+     * @param setpoint desired angle in degrees
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/pitcher/PitcherSS.java:39]: the desired

File: src/main/java/frc/robot/subsystems/loader/LoaderSS.java [hunk 20]
```diff
@@ -20,15 +20,15 @@ public void move(double power) {
     }
 
     /**
-     * @param velocity desired velocity in RPM
+     * @param setpoint desired velocity in RPM
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/loader/LoaderSS.java:23]: the desired

File: src/main/java/frc/robot/subsystems/intake/IntakeOpenerSS.java [hunk 21]
```diff
@@ -35,14 +35,14 @@ public double getAngle() {
     /**
      * Sets the intake to the given angle
      *
-     * @param degree desired angle in degrees
+     * @param setpoint desired angle in degrees
```
Comment (laviRZ) [inline src/main/java/frc/robot/subsystems/intake/IntakeOpenerSS.java:38]: the desired

**Review by laviRZ — APPROVED** [review]
change and merge

**Pull request merged**